### ISSUE TITLE
Should streams handle triggering reducers themselves?

### ISSUE THREAD

**Issue by Aleksion** [Aleksion/reduxSwift#11]
I noticed that with the implementation of streamFactories you changed how the reducers were triggered.

I the original implementation this was done by RxSwift as such:

``` swift
    // have the stateSubject subscribe to the dispatcher
    subjectDispatcher
        .scan(currentState, accumulator: reducer)
        .startWith(currentState)
        .subscribe(stateSubject)
```

So it used observables to trigger reducers.
In the updated version, you create a dispatch function - in the createStreamStore function -  which calls stream.dispatch with an explicit call to the reducer as a parameter

``` swift
    func dispatch(action: Action) -> Action {
        stream.dispatch(reducer(stream.getState(), action))
        return action
    }
```

Should each individual stateStream handle how they trigger reducers themselves - or should we keep doing it manually in the createStreamStore function? 
Or should we simply pass the stream.dispatch function directly to the store and let the implementation details be handled in the stream implementation?
I would like to document that one a bit better, so newcomers can understand how data flows through the framework. I can do that when we've decided on how to do it best.

**Comment by agentk** [comment]
I went back and forth a few time before settling on the store doing the reducing. What I found surprising was, using accumulators, reducers or subscribers from the RP frameworks was twice as slow as the explicit reduce in the performance test when dispatching 1000 IncrementActions.
I know it's not indicative of where cpu cycles are spent in real world applications.
The createStore function is pretty simple if we wanted one that pushed the reduce responsibility back to the framework.
Probably the only caution for users creating their own store is not to use a cold signal as it wont pass actions to the reducer until it's observed.
